### PR TITLE
Dmirano/3979 errors warning cleanup

### LINKED ISSUE #3973: [Tech Debt]: Cleanup errors/warning in console for the front end app
While none of the errors/warnings we are seeing in the console appear to have any impact on the application functionality, it's generally good "code hygiene" to keep these to a minimum or keep them at zero. 

We should take some time to go through, find them all, and remedy them. 

### This task is done when...

- [ ] All console errors and warnings addressed
- [ ] criteria 2

---

Add additional labels (design, dev, compliance, BUG, etc) and size before submitting.

If the issue is needed to complete prioritized work for the CURRENT SPRINT, add it to the "This Sprint" pipeline. Otherwise, all other issues will be automatically added to the _unprioritized pipeline_ for prioritization in backlog refinement or sprint planning with Product.

### PR DESCRIPTION
**Pull request by mirano-darren**
Resolves #3973 

### Description

Addresses come console warning throughout the app. Most of which were warnings related to prop types

### Signifcant changes or possible side effects
Added another type of acceptable prop type where warnings appeared. 

### Automated test cases written
None

### Steps to manually verify this change

1. frontend tests all pass
2. cypress tests pass
3. Nothing visually changed, so nothing really to manually verify. 

### This pull request is ready to review when

- [x] Automated tests are updated (and all tests are passing)
- [x] Pull request has been labeled, if applicable with feature, content, bug, tests, refactor
- [ ] Associated OpenAPI documentation has been updated
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/pages/apd/activities/oms/OutcomesAndMetrics/OutcomeAndMetricReview.js [hunk 1]
```diff
@@ -19,16 +19,16 @@ const OutcomeAndMetricReview = ({
       {outcome || 'Outcome not specified'}
     </p>
     {metrics.length > 0 ? (
-      <p className="ds-u-margin-top--2">
-        <ul className="ds-c-list--bare subform__container">
+      <ul className="ds-c-list--bare subform__container">
```
Comment (tbolt) [inline web/src/pages/apd/activities/oms/OutcomesAndMetrics/OutcomeAndMetricReview.js:22]: This isn't valid html (it wasn't originally either 🤣)

html list elements `<ul>` or `<ol>` should only contain `<li>` as its direct children.

so id recommend updating the jsx to reflect this structure-
```
<div class="subform__container ds-u-margin-top--2">
  <strong>Metrics:</strong>
  <ul class="ds-c-list--bare">
    <li class="ds-u-margin-bottom--2">[item 1]</li>
    <li class="ds-u-margin-bottom--2">[item 2]</li>
  </ul>
</div>
```

**Review by tbolt — COMMENTED** [review]
Added a note for a (hopefully) easy update/fix

File: web/src/pages/apd/activities/oms/OutcomesAndMetrics/OutcomeAndMetricReview.js (hunk 1 above)
Comment (mirano-darren) [reply web/src/pages/apd/activities/oms/OutcomesAndMetrics/OutcomeAndMetricReview.js:22]: 😆 nice catch!

**Review by thetif — APPROVED** [review]
looks good

**Pull request merged**